### ISSUE TITLE
Spell checking tool

### ISSUE THREAD

**Issue by leventov** [apache/incubator-druid#7209]
Introduce a tool to catch spelling and common typing problems, such as "the the", "decomission", etc.

In the meantime, we can use Checkstyle's Regexp pattern, but that's pretty inefficient(?) and may slow down the build.

Inspired by #7154.

**Comment by gianm** [comment]
I'm not sure if a generic spell check is worth it -- there might be way too many false positives and it would just get annoying -- but something that looks only for _specific_ misspellings may be worth it. (Although, of limited usefulness, since it wouldn't be checking very much.)

I'd suggest doing something that can be applied through Maven (rather than IntelliJ inspections) since it's more accessible. Everyone working on Druid has Maven, not everyone has IntelliJ, and it's a better experience when people can verify their own stuff before submitting it up as a PR. The IntelliJ inspection errors won't be obvious until TC runs, which introduces delay in the PR cycle.

**Comment by gianm** [comment]
Btw, did you just mean docs? I was figuring this suggestion was about adding a spell checker specifically for docs, but just realized the top comment didn't mention that.

**Comment by leventov** [comment]
> I'm not sure if a generic spell check is worth it -- there might be way too many false positives and it would just get annoying -- but something that looks only for specific misspellings may be worth it. (Although, of limited usefulness, since it wouldn't be checking very much.)

> I'd suggest doing something that can be applied through Maven (rather than IntelliJ inspections) since it's more accessible. Everyone working on Druid has Maven, not everyone has IntelliJ, and it's a better experience when people can verify their own stuff before submitting it up as a PR. The IntelliJ inspection errors won't be obvious until TC runs, which introduces delay in the PR cycle.

Yes, that's exactly what I meant - a Maven plugin which is feeded with a list of stop words (BTW we can include profanity there, too). My only concern with Checkstyle's Regexp rules is that they are regular expressions and having a lot of them to check in the whole repo may slow down the build. The stop word plugin should do simple exact match scan.
